glib/src/translate.rs

```diff
@@ -1229,6 +1253,50 @@ impl FromGlib<ffi::gboolean> for bool {
     }
 }
 
+impl FromGlib<ffi::gpointer> for Pointer {
```

Comment:
Are these trait impls useful for something?

Reply:
It's partially to be consistent/interchangeable with `NonNull`, but more important when you want to be generic over the pointer types. It's not strictly needed until [you start changing some of these into blanket impls](https://github.com/arcnmx/gtk-rs-core/commit/pointer-subclass#diff-13ade07b8d0274c89030bc91feea353311f502b1d30976ae8c9f37814851779bL837), though.

Reply:
But that also only works for `c_void`-typed pointer and for anything else you'd still need to cast and do things manually. I'm mostly wondering how useful that is in practice, can it actually be used in any meaningful way in real code?

Reply:
Hm, well any type that interacts with `g_value_get_pointer` must be castable to/from `gpointer` in some way. These traits seemed the most appropriate, but because `FromGlib<>` is parameterized, you need a concrete instantiation of it like `FromGlib<gpointer>` to use as a common interface.

> I'm mostly wondering how useful that is in practice, can it actually be used in any meaningful way in real code?

Which parts specifically? Sure, it's more useful with the associated blanket impls (that give the same flexibility to all other pointer types, and allow you to newtype them, etc), but the ability to use optional/non-null pointers is still quite useful if they're untyped!

> for anything else you'd still need to cast and do things manually

Blanket impls allow you to skip casting, but I believe there's still use in modelling them more specifically than just a plain `gpointer`, at which point they desired type is just a `.cast::<T>()` away. Often whether a pointer is null or not is relevant to code before it actually needs to work with the type! In other words code often wants to deal with the `None` case in a way where the pointed type itself is irrelevant - then it passes the NonNull (typed or untyped, casted or not) to further logic for actual processing.
One of my goals here is to enable `gir` to deal with pointers better, and be able to model nullable vs non-nullable pointers the same way it can currently model object types. For example something like the following generated code should "just work":
```rust
fn user_data(&self) -> Option<NonNull<Pointee>> {
  unsafe {
    from_glib(ffi::x_get_user_data(self.to_glib_none().0)
  }
}
```

---------

I'll go ahead and kill any extraneous ones for now - it's not like I can't just write `NonNull::new()` instead of `from_glib`. I'll pull the necessary ones back in as part of subclassing and after further experimentation with gpointers in the library I'm wrapping.

For what it's worth, I will be writing code where `value.get::<NonNull<f32>>()` makes sense to want, so I do need to flesh these impls out in interesting ways... But one could argue that the `c_void` variant of `value.get::<NonNull<Pointee>>()?.cast::<f32>()` is more correct, because the type `*mut f32` isn't actually represented by the `value.type_()`.